internal/resources/flush_bucket.go

```diff
@@ -0,0 +1,186 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/api"
+	"github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/errors"
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &FlushBucket{}
+	_ resource.ResourceWithConfigure   = &FlushBucket{}
+	_ resource.ResourceWithImportState = &FlushBucket{}
+)
+
+// FlushBucket is the bucket resource implementation.
+type FlushBucket struct {
+	*providerschema.Data
+}
+
+// NewFlushBucket is a helper function to simplify the provider implementation.
+func NewFlushBucket() resource.Resource {
+	return &FlushBucket{}
+}
+
+// Metadata returns the flush Bucket resource type name.
+func (c *FlushBucket) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_flush"
+}
+
+// Schema defines the schema for the flush Bucket resource.
+func (c *FlushBucket) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = FlushBucketSchema()
+}
+
+// Create creates a new flush Bucket.
+func (c *FlushBucket) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.FlushBucket
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if err := c.validateFlushBucketRequest(plan); err != nil {
+		resp.Diagnostics.AddError(
+			"Error executing bucket flush",
+			"Could not flush the bucket, unexpected error: "+err.Error(),
+		)
+		return
+	}
+	var organizationId = plan.OrganizationId.ValueString()
+	var projectId = plan.ProjectId.ValueString()
+	var clusterId = plan.ClusterId.ValueString()
+	var bucketId = plan.BucketId.ValueString()
+
+	// Execute flush bucket. Nothing gets returned for it.
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/buckets/%s/flush", c.HostURL, organizationId, projectId, clusterId, bucketId)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPut, SuccessStatus: http.StatusOK}
+	_, err := c.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		c.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error flushing the bucket",
+			errorMessageWhileBucketCreation+api.ParseError(err),
```

Comment:
This error is confusing to users.  The detail message will still suggest error during bucket creating.  Please create new error message.